test/e2e/runner/setup.go

```diff
@@ -179,6 +179,10 @@ func MakeConfig(node *e2e.Node) (*config.Config, error) {
 	cfg.BlockSync.Version = node.BlockSyncVersion
 	cfg.Consensus.PeerGossipIntraloopSleepDuration = node.Testnet.PeerGossipIntraloopSleepDuration
 
+	if node.Mode == e2e.ModeValidator || node.Mode == e2e.ModeFull {
+		cfg.Storage.Pruning.DataCompanion.Enabled = true
+	}
```

Comment:
I noticed that we didn't even enable the `storage.pruning.data_companion.enabled` flag for the E2E tests, so they were incorrectly passing.